src/modules/cpu/kernel/normalize_generic.hpp

```diff
@@ -0,0 +1,513 @@
+            {
+                Rpp32u reductionDims = 2 * (batchCount + 1);
+                meanTensor = (Rpp32f *)calloc(length[reductionDims], sizeof(Rpp32f));
+                stdDevTensor = (Rpp32f *)calloc(length[reductionDims], sizeof(Rpp32f));
+                srcStride[0] = srcStride[1] = srcGenericDescPtr->strides[2];
+                srcReductionDims[0] = 1;
+                srcReductionDims[1] = srcAudioDims[0] * srcAudioDims[1];
+                paramStride[0] = paramStride[1] = 0;
+            }
+            else if (axis_mask == 1)
+            {
+                Rpp32u reductionDims = 2 * (batchCount + 1);
+                meanTensor = (Rpp32f *)calloc(length[reductionDims], sizeof(Rpp32f));
+                stdDevTensor = (Rpp32f *)calloc(length[reductionDims], sizeof(Rpp32f));
+                srcStride[0] = srcGenericDescPtr->strides[1];
+                srcStride[1] = srcGenericDescPtr->strides[0];
+                srcReductionDims[0] = srcAudioDims[1];
+                srcReductionDims[1] = srcAudioDims[0];
+                paramStride[0] = 1;
+                paramStride[1] = 0;
+            }
+            else if (axis_mask == 2)
+            {
+                Rpp32u reductionDims = batchCount * 2 + 1;
+                meanTensor = (Rpp32f *)calloc(length[reductionDims], sizeof(Rpp32f));
+                stdDevTensor = (Rpp32f *)calloc(length[reductionDims], sizeof(Rpp32f));
+                srcStride[0] = srcGenericDescPtr->strides[0];
+                srcStride[1] = srcGenericDescPtr->strides[1];
+                srcReductionDims[0] = srcAudioDims[0];
+                srcReductionDims[1] = srcAudioDims[1];
+                paramStride[0] = 0;
+                paramStride[1] = 1;
+            }
+            if(computeMean)
+                compute_2D_mean(srcPtrTemp, meanTensor, srcReductionDims, srcStride);
+            if(computeStddev)
+                compute_2D_inv_std_dev(srcPtrTemp, meanTensor, stdDevTensor, srcReductionDims, srcStride);
+
+            // Inv std dev calculations missing
+        if(axis_mask == 2)
+            normalize_2D_tensor_avx_axis2(srcPtrTemp, srcGenericDescPtr, dstPtrTemp, dstGenericDescPtr, meanTensor, stdDevTensor, shift, srcAudioDims, paramStride);
+        else
+            normalize_2D_tensor(srcPtrTemp, srcGenericDescPtr, dstPtrTemp, dstGenericDescPtr, meanTensor, stdDevTensor, shift, srcAudioDims, paramStride);
+        }
+        else if(nDim == 3)
+        {
+            if(axis_mask == 1) // Normalize across Channels
+            {
+                paramStride[0] = paramStride[1] = 1;
+                paramStride[2] = 0;
+            }
+            else if(axis_mask == 2) // Normalize across Width
+            {
+                //TODO
+            }
+            else if(axis_mask == 3) // Normalize across Height
+            {
+                //TODO
+            }
+            srcPtrChannel = srcPtrTemp + (begin[0] * layoutParams.bufferMultiplier);
```

Comment:
Why this is needed?
Please elaborate on this

Reply:
What if our ROI begin value differs, right now we send in 0 so that okay but we need to modify and process as per roi pointers too right.